build.rs

```diff
@@ -0,0 +1,85 @@
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use maplit::hashmap;
+use memofs::VfsSnapshot;
+
+fn snapshot_from_fs_path(path: &PathBuf) -> Result<VfsSnapshot, PathBuf> {
+    if path.is_dir() {
+        let entries: Result<Vec<fs::DirEntry>, PathBuf> = fs::read_dir(path)
+            .map_err(|_| path.to_owned())?
+            .map(|entry| entry.map_err(|_| path.to_owned()))
+            .into_iter()
+            .collect();
+
+        let vfs_entries: Result<Vec<(String, VfsSnapshot)>, PathBuf> = entries?
+            .iter()
+            .filter(|entry| {
+                let file_name = entry.file_name();
+                !file_name.to_string_lossy().ends_with(".spec.lua")
+            })
+            .map(|entry| {
+                let path = entry.path();
+
+                entry
+                    .file_name()
+                    .to_str()
+                    .ok_or(path.to_owned())
+                    .and_then(|file_name| {
+                        snapshot_from_fs_path(&path)
+                            .map(|snapshot| (file_name.to_owned(), snapshot))
+                    })
+            })
+            .into_iter()
+            .collect();
```

Comment:
Two notes here:

1. The result of `map` is already an `Iterator`, so we don't need to call `into_iter` on it.
2. `VfsSnapshot::dir` accepts an iterator as an argument, so there's no need to collect here.